### PR TITLE
emulation: save state after gscond

### PR DESCRIPTION
**Pull request by nbren12**
The temperature and humidity states after gscond are used to compute a relative
humidity tendency in the following timestep. Also, we always run the model with
the same physics (dtp) and dynamics (dtf) timestep, so the tp1 and cp1 variables
are identical to cp. dtp is actually set equal to dtf in the model startup.
See this
[notebook](https://github.com/ai2cm/explore/blob/master/noahb/emulation/microphysics/2021-11-03-verify%20tbd%20data.ipynb)
for proof.

* Remove the redundnant call py fort state setting for tp1_cpf, etc
* Rename the output of `tp` etc to be more accurate
* Save the temperature and humidity state after `gscond`
* Install black in nix environment to make development easier

### CODE REVIEW

**Comment by nbren12** [comment]
See https://github.com/ai2cm/fv3net/issues/1491 for more background.

File: FV3/gfsphysics/GFS_layer/GFS_physics_driver.F90
Comment (frodre) [inline FV3/gfsphysics/GFS_layer/GFS_physics_driver.F90:4590]: Not sure I understand how the precip works and want to make sure the land model has what it needs.  I see a diagnostic set right after physics runs, but shouldn't we leave the `call get_state("total_precipitation", rain1)`?  Otherwise we don't have a way of updating with the emulated value.  Let me know if that's not the case.

File: tests/emulation/emulation/emulate.py
Comment (frodre) [inline tests/emulation/emulation/emulate.py:7]: 👍🏼👍🏼

**Review by frodre — COMMENTED** [review]
Thanks for the update @nbren12.  A few things.

* I think the `get_state` emulator update to `total_precipitation` was accidentally deleted
* Microphysics emulation naming is currently pretty coupled to some of the code in fv3net that is used for training
    * Using `_input` `_output` suffixes to get tendency values in [derived mapping](https://github.com/ai2cm/fv3net/blob/ce183c161999e07505b2d3951635ce64f850389e/external/vcm/vcm/derived_mapping.py#L258)
    * Attributes for certain fields in the monitor/netcdf output are [also tied to the `_input` `_output` naming](https://github.com/ai2cm/fv3net/blob/ce183c161999e07505b2d3951635ce64f850389e/external/emulation/emulation/_monitor/monitor.py).  This would only affect the availability of metadata on saved fields via these routines, and there are already fields in the state which dont' have any metadata (so it wouldn't be a new problem)

We should resolve when updating into `fv3net`.

The `total_precip` question is the only thing that needs to be addressed before approval.

File: FV3/gfsphysics/GFS_layer/GFS_physics_driver.F90
Comment (nbren12) [reply FV3/gfsphysics/GFS_layer/GFS_physics_driver.F90:4611]: >  I was thinking that letting the physics do the previous timestep state updates would be sufficient,

I am still not convinced this is the problem, but now we can actually tell.

File: FV3/gfsphysics/GFS_layer/GFS_physics_driver.F90
Comment (nbren12) [reply FV3/gfsphysics/GFS_layer/GFS_physics_driver.F90:4590]: Good catch! I didn't intend to delete this.

File: tests/emulation/emulation/emulate.py
Comment (nbren12) [reply tests/emulation/emulation/emulate.py:7]: :grin:

**Comment by nbren12** [comment]
@frodre Thanks for this todo list:

>- Attributes for certain fields in the monitor/netcdf output are also tied to the _input _output naming. This would only affect the availability of metadata on saved fields via these routines, and there are already fields in the state which dont' have any metadata (so it wouldn't be a new problem)

Would moving the metadata definitions into fv3gfs-fortran help this? Seems like the metadata should be defined as close to the code as possible.

>- Using _input _output suffixes to get tendency values in derived mapping

We could probably make this code a bit more decoupled within fv3net.

**Comment by frodre** [comment]
> Would moving the metadata definitions into fv3gfs-fortran help this? Seems like the metadata should be defined as close to the code as possible.

The metadata being in fv3net was mostly about the the emulation modules also being there.  Since the included state fields in this repo is fairly static, I think we can leave it there and just make sure it gets updated.  It's not a breaking change since it just doesn't include info it doesn't find.

**Review by frodre — APPROVED** [review]
Thanks for checking into the precip variable update.

**Review by elynnwu — APPROVED** [review]
Looks good to me!

**Pull request merged**
